src/daml/Daml/Trading/Reward/TradingPoint.daml

```diff
@@ -1 +1,72 @@
 module Daml.Trading.Reward.TradingPoint where 
+
+template TradeSuccess
+  with
+    admin : Party
+    seller : Party
+    buyer : Party
+    tradeAmount : Decimal
+    asset : Text
+  where 
+    signatory admin
+    observer seller, buyer
+    ensure seller /= buyer
+    choice IssueRewardPoint : (ContractId TradingPoint, ContractId TradingPoint)
+      controller admin
+      do  
+        (sellerTradingPointCid, sellerTradingPoint) <- fetchByKey @TradingPoint (admin, seller)
+        (buyerTradingPointCid, buyerTradingPoint) <- fetchByKey @TradingPoint (admin, buyer)
+        let    
+          calArguments : Decimal -> (Int, Decimal)
```

Comment:
We should put this function outside the template.
Also please use a better name than `calArguments`